src/main/java/org/iitbact/cc/services/ReportService.java

```diff
@@ -0,0 +1,75 @@
+package org.iitbact.cc.services;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.iitbact.cc.entities.Facility;
+import org.iitbact.cc.entities.Patient;
+import org.iitbact.cc.repository.FacilityRepository;
+import org.iitbact.cc.repository.PatientRepository;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ReportService {
+
+	private final FacilityRepository facilityRepository;
+	private final PatientRepository patientRepository;
+
+	public ReportService(FacilityRepository facilityRepository,
+			PatientRepository patientRepository) {
+		this.facilityRepository = facilityRepository;
+		this.patientRepository = patientRepository;
+	}
+
+	public List<String> fetchFacilityData() {
+		List<Facility> facilities = facilityRepository.findAll();
+
+		// Add header for the CSV file.
+		List<String> resultSetArray = new ArrayList<>();
+		resultSetArray.add("ID,Name,Area,Facility Type,Telephone,Email");
+
+		// Add the data rows.
+		for (Facility f : facilities) {
+			String line = String.join(",", Integer.toString(f.getFacilityId()), f.getName(),
+							f.getArea(), f.getCovidFacilityType(), f.getTelephone(), f.getEmail());
+			resultSetArray.add(line);
+		}
+		return resultSetArray;
+
+	}
+
+	public List<String> fetchPatientData() {
+		List<Patient> patients = patientRepository.findAll();
+
+		// Add header for the CSV file.
+		List<String> resultSetArray = new ArrayList<>();
+		resultSetArray.add(
+				"ID,GOI Covid ID,Name,Age,Gender,Address,Locality,Pincode,Occupation,Contact Number,District,State");
+
+		// Add the data rows.
+		for (Patient p : patients) {
+			String line = String.join(",", Integer.toString(p.getPatientId()), p.getGoiCovidId(),
+					p.getName(), Integer.toString(p.getAge()), p.getGender(), p.getAddress(),
```

Comment:
This will breakdown if name, address or any other field contains a comma. There is a high chance that address will contain commas.
We should use CSVWriter instead of FileWriter. That takes care of all such things.

Reply:
Yes, good observation. This is mostly a placeholder right now, and using CsvWriter is on my TODO list for tonight.

Reply:
FYI - Replaced FileWriter with CsvWriter in this new commit - https://github.com/adminiitbact/covid.control.backend/commit/cdfe52b2aa5d21a00887880f91f1b59071cb2e7a